src/azul/portal_service.py

```diff
@@ -172,12 +172,12 @@ def _crud(self, operation: Callable[[JSONs], Optional[JSONs]]) -> Optional[JSONs
                     except NoSuchObjectVersion:
                         # Wait for latest version to appear in S3.
                         continue
-                db = operation(db)
-                if db is None:
+                db_op = operation(db)
+                if db_op is None:
                     # Operation is read-only; we're done.
                     break
                 else:
-                    self._write_db(db, version)
+                    self._write_db(db_op, version)
```

Comment:
For write operations, `_crud` currently returns the version of the database created by the write, while this changes it to return the previous version (i.e. the version that was read before being overwritten). I think it would be better to preserve the current behavior, which I think can easily be accomplished by:

```suggestion
                    self._write_db(db_op, version)
                    db = db_op
```